### PR TITLE
Fix staticcheck issues

### PR DESCRIPTION
**Pull request by mjtrangoni**
Hi @dougm,

I fixed some [staticcheck](https://github.com/dominikh/go-tools) issues here. If you agree with this, I can fix the rest. This check should also be added to TravisCI in the future.

### CODE REVIEW

**Comment by mjtrangoni** [comment]
@dougm PTAL. I dropped the `vim25/xml` changes.

File: govc/vm/info.go [hunk 1]
```diff
@@ -101,12 +101,10 @@ func (cmd *info) Run(ctx context.Context, f *flag.FlagSet) error {
...
-	if err != nil {
-		if _, ok := err.(*find.NotFoundError); ok {
-			// Continue with empty VM slice
-		} else {
-			return err
-		}
+	if _, ok := err.(*find.NotFoundError); ok {
+		// Continue with empty VM slice
+	} else {
+		return err
```
Comment (dougm) [inline govc/vm/info.go]: We don't want to return `err` here if it is `nil`, looks like this is what's causing CI to fail - at least if fails locally for me unless this hunk is reverted.

Comment (mjtrangoni) [reply govc/vm/info.go]: @dougm Good catch! I reverted it for the moment. Thanks!

**Comment by mjtrangoni** [comment]
@dougm PTAL. I fixed more trivial issues. 
The current status is, 
```
$ staticcheck ./... | wc -l 
40
```

**Pull request merged**